tests/integration/providers/ldap/delta_outbound_sync_tests.py

```diff
@@ -0,0 +1,102 @@
+# Copyright 2018 Contributors to Hyperledger Sawtooth
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# -----------------------------------------------------------------------------
+"""LDAP Delta Outbound Sync Tests"""
+# pylint: disable=redefined-outer-name
+# NOTE: disabling for pytest as per:
+# https://stackoverflow.com/questions/46089480/pytest-fixtures-redefining-name-from-outer-scope-pylint
+
+import pytest
+
+from ldap3 import Connection, MOCK_SYNC, OFFLINE_AD_2012_R2, Server
+
+from rbac.common.logs import get_default_logger
+from rbac.providers.common.inbound_filters import inbound_group_filter
+from rbac.providers.ldap.delta_outbound_sync import create_entry_ldap
+from tests.integration.providers.ldap.delta_inbound_sync_tests import (
+    create_fake_user,
+    get_fake_group,
+    get_fake_user,
+    put_in_inbound_queue,
+)
+
+LOGGER = get_default_logger(__name__)
+SERVER = Server("my_fake_server", get_info=OFFLINE_AD_2012_R2)
+
+
+@pytest.fixture(autouse=True, scope="module")
+def ldap_connection():
+    """Binds and yields a mock ldap connection for integration testing."""
+    connection = Connection(
+        SERVER,
+        user="cn=my_user,ou=test,o=lab",
+        password="my_password",
+        client_strategy=MOCK_SYNC,
+    )
+    connection.bind()
+    yield connection
+    connection.unbind()
+
+
+def test_create_new_role_outbound(ldap_connection):
+    """ Tests LDAP role creation through getting outbound_queue entry and processing
+    the outbound_queue entry.
```

Comment:
so this is testing that a role ingested through the inbound queue is being inserted into the outbound queue with correct formatting?

Reply:
No - this is testing whether or not the LDAP Outbound Sync can process correctly formatted outbound_queue entries.

I created a fake user and inserted it into NEXT because I needed to mimic that an imported LDAP user created a role on NEXT. 

At the end, I query what's in LDAP to make sure the group was created properly.

Reply:
cool, do you mind changing the docstring to something like 
>testing whether or not the LDAP Outbound Sync can process correctly formatted outbound_queue entries.

? 
the original one was a little confusing

Reply:
Completed.